go-controller/pkg/util/kube.go

```diff
@@ -332,6 +333,22 @@ func StartNodeCertificateManager(ctx context.Context, wg *sync.WaitGroup, nodeNa
 		}
 	}
 
+	// In the unlikely event that the certificate file becomes corrupted, recover by removing
+	// the certificate so the CSR will be created using the bootstrap kubeconfig.
+	var noCertKeyError *certificate.NoCertKeyError
+	if err != nil && !errors.As(err, &noCertKeyError) {
+		var pathErr *os.PathError
+		klog.Errorf("The certificate file is corrupted: %v", err)
```

Comment:
nit
By the description of what you are doing in the comment, it feels like this log should be below the check for PathError (I guess there is potentially other reasons for a PathError) or say something more generic ("Failed to read the certificate file"?)